### DISCUSSION TITLE
Livewire newbie

### DISCUSSION THREAD

**Discussion by luisetto6** [livewire/livewire#3488] [1. Help] [answered]
Hi people,

Newbie here, learning some laravel to develop an e-commerce website following a series of videos from youtube. I'm stuck with a problem that I can't figure it out where do I have the issue, here's the series I'm following:

https://www.youtube.com/watch?v=nzF2ARPF2HE&list=PLz_YkiqIHesvPtvLl2Wz5FtuW44dBt199&index=14&ab_channel=SurfsideMedia

I'm not using the same template, but the code is almost the same for the blade components. Right now my main issue is that I cannot create a category, for some reason the generateSlug function is not working and if I type it manually and press submit doesn't save it on the DB. I'm going to share my Model and my components and blades:

Category Model
```php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Category extends Model
{
    use HasFactory;

    protected $table = "categories";

}
```

Here's my AdminAddCategoryComponent:

```php
<?php

namespace App\Http\Livewire;

use App\Models\Category;
use Livewire\Component;
use Illuminate\Support\Str;

class AdminAddCategoryComponent extends Component
{
    public $name;
    public $slug;

    public function generateSlug(){

          $this->slug = Str::slug($this->name,'-');
    
    }

       
    public function addCategory(){

        $categ = new Category();
        $categ->name = $this->name;
        $categ->slug = $this->slug;
        $categ->save();
        session()->flash('message','Category Added Successfully');

    }
    
    public function render()
    {
        return view('livewire.admin-add-category-component')->layout('layouts.base');
    }
}
```

Here's my blade Admin-Add-Category:

```html
<div class="panel-body">
                    @if (Session::has('message'))
                        <div class="alert alert-success" role="alert">{{Session::get('message')}}</div>
                    @endif
                    <form class="form-horizontal" wire:submit.prevent="addCategory">
                        <div class="form-group">
                            <label class="col-md-4 control-label">Category Name</label>
                            <div class="col-md-4">
                                <input type="text" placeholder="Category Name" class="form-control input-md" wire:model="name" wire:keyup="generateSlug"/>
                               
                            </div>
                        </div>

                        <div class="form-group">
                            <label class="col-md-4 control-label">Category Slug</label>
                            <div class="col-md-4">
                                <input type="text" placeholder="Category Slug" class="form-control input-md" wire:model="slug"/>
                                
                            </div>
                        </div>

                        <div class="form-group">
                            <label class="col-md-4 control-label"></label>
                            <div class="col-md-4">
                                <button type="submit" class="btn btn-primary">Submit</button>
                            </div>
                        </div>
                    </form>

                </div>```

I'm kind of frustrated because I cannot seem to find out what's happening and I have been checking it out for a couple of days... Any help would be appreciated
Regards
Luis
[AdminAddCatComponent.txt](https://github.com/livewire/livewire/files/6950238/AdminAddCatComponent.txt)
[blade.txt](https://github.com/livewire/livewire/files/6950239/blade.txt)
[model.txt](https://github.com/livewire/livewire/files/6950240/model.txt)

**Comment by binaryweavers** [comment]
@luisetto6 I was in mood of help, but question was hard to understand. Here is the question in proper formatting. Now let me check what's the issue If I could find it
_____________________________________________________________________________________________________________________________________________
Hi people,

Newbie here, learning some laravel to develop an e-commerce website following a series of videos from youtube. I'm stuck with a problem that I can't figure it out where do I have the issue, here's the series I'm following:

https://www.youtube.com/watch?v=nzF2ARPF2HE&list=PLz_YkiqIHesvPtvLl2Wz5FtuW44dBt199&index=14&ab_channel=SurfsideMedia

I'm not using the same template, but the code is almost the same for the blade components. Right now my main issue is that I cannot create a category, for some reason the generateSlug function is not working and if I type it manually and press submit doesn't save it on the DB. I'm going to share my Model and my components and blades:

`App\Models\Category`
```php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Category extends Model
{
use HasFactory;

protected $table = "categories";
}
```

`AdminAddCategoryComponent:`
```php
<?php

namespace App\Http\Livewire;

use App\Models\Category;
use Livewire\Component;
use Illuminate\Support\Str;

class AdminAddCategoryComponent extends Component
{
    public $name;
    public $slug;

    public function generateSlug(){

          $this->slug = Str::slug($this->name,'-');
    
    }

       
    public function addCategory(){

        $categ = new Category();
        $categ->name = $this->name;
        $categ->slug = $this->slug;
        $categ->save();
        session()->flash('message','Category Added Successfully');

    }
    
    public function render()
    {
        return view('livewire.admin-add-category-component')->layout('layouts.base');
    }
}

```
`blade`

```blade
<div class="panel-body">
    @if (Session::has('message'))
        <div class="alert alert-success" role="alert">{{ Session::get('message') }}</div>
    @endif
    <form class="form-horizontal" wire:submit.prevent="addCategory">
        <div class="form-group">
            <label class="col-md-4 control-label">Category Name</label>
            <div class="col-md-4">
                <input type="text" placeholder="Category Name" class="form-control input-md" wire:model="name" wire:keyup="generateSlug" />

            </div>
        </div>

        <div class="form-group">
            <label class="col-md-4 control-label">Category Slug</label>
            <div class="col-md-4">
                <input type="text" placeholder="Category Slug" class="form-control input-md" wire:model="slug" />

            </div>
        </div>

        <div class="form-group">
            <label class="col-md-4 control-label"></label>
            <div class="col-md-4">
                <button type="submit" class="btn btn-primary">Submit</button>
            </div>
        </div>
    </form>

</div>

```

I'm kind of frustrated because I cannot seem to find out what's happening and I have been checking it out for a couple of days... Any help would be appreciated 
Regards Luis

**Reply by luisetto6** [reply]
Hi!! 

Sorry por the bad formatting, thanks for fixing it and for try to help me out..

**Comment by joshhanley** [comment]
@luisetto6 when you paste code into github, you need to surround it in triple back ticks and then specify the file type like php or blade.

For example
\```php
\\ PHP code here
\```

\```blade
\\ HTML and blade code here
\```

In regards to your issue, you need to make sure your methods are actually being called.

Inside your `generateSlug` and `addCategory` methods, put a dd dump in each of them.

So for `generateSlug` put at the top `dd('generateSlug');`.

And for `addCategory` put at the top `dd('addCategory');`.

Then test it out again. When you type name, you should see "generateSlug" pop up, and if you click on your "submit" button you should see "addCategory".

Can you test that and let us know if that is working and you see the dumps?

**Reply by luisetto6** [reply]
Hi Josh,

Thanks a lot for your response, I fixed the formatting with your recommendations and now it seems to be readable!

I followed your recommendations and putted a dd in both methods, nothing happended.. I thought I was doing it wrong and went to another form that it was working well and putted the DD and it showed a popup screen showing the function it was using..

I doubled checked the names to see if there were any typos, created another component with different routes and still showing the same issue..

Any idea what else can I check? 

Thanks in advance for your recommendations

**Reply by binaryweavers** [reply]
Hello @luisetto6  sorry for late response I couldn't make to my machine after that.
Can you try typing `window.livewire` or `window.Livewire` in browser console on the page you having issue. Just to check if livewire js is available at all
You should see something like this
![image](https://user-images.githubusercontent.com/19465738/128794048-5d45518c-9cac-4677-8f8f-24ba32e54c78.png)

**Reply by luisetto6** [reply]
Hi @binaryweavers 

Here's the output when I type the command:

![image](https://user-images.githubusercontent.com/64087331/128795329-a55014b5-0105-48c1-a9c2-e99066c62343.png)

**Reply by binaryweavers** [reply]
This seems fine. I see no issue in component class and blade file as well 🤔

**Reply by luisetto6** [reply]
Any command I can use to like erase cache? I have to put "php artisan route: clear" everytime I create a new component and routes because for some reason it shows a 404 error if I don't.. Maybe it's something like that..

**Comment by joshhanley** [comment]
@luisetto6 another thing you can do, is open your browser dev tools and go to the network tab and filter by "Fetch/XHR".

Then either update your name field or click on your button.

You should see an XHR request pop up, click on that, then click on the "Headers" tab.

Scroll all the way to the bottom, to the "Request Payload" section, and expand "fingerprint" and "updates" and take a screenshot of all of it.

Here is an example of what it should look like

![image](https://user-images.githubusercontent.com/882837/128834075-f82cb05e-06db-42bf-940e-55ff01cc39db.png)

**Reply by luisetto6** [reply]
Hi Josh,

Thanks for your recommendation, I went to dev tools and network option and filtering by XHR and nothing showed up, I have two forms that are presenting this issue.. Went to another form that is working well and it calls the livewire component:

![image](https://user-images.githubusercontent.com/64087331/128921929-7b733b25-b4d8-4ff6-ae84-e3e1301182ba.png)

Here's the one that doesn't work:

![image](https://user-images.githubusercontent.com/64087331/128922087-17669e32-da19-4e48-a6ab-bd2348261601.png)

**Comment by luisetto6** [comment, marked as answer]
@joshhanley @binaryweavers  Hi guys, a friend helped me out with this.. I had son br tags in my blade that were affecting.. Now it's working. Thanks again for your patience and help

**Reply by joshhanley** [reply]
No worries, glad you got it sorted 🙂
